src/session/session_api.c

```diff
@@ -1682,6 +1682,32 @@ __session_query_timestamp(WT_SESSION *wt_session, char *hex_timestamp, const cha
     API_END_RET(session, ret);
 }
 
+/*
+ * __session_reset_snapshot --
+ *     WT_SESSION->reset_snapshot method.
+ */
+static int
+__session_reset_snapshot(WT_SESSION *wt_session)
+{
+    WT_SESSION_IMPL *session;
+
+    session = (WT_SESSION_IMPL *)wt_session;
+    /* Return error if the isolation mode is read committed. */
+    if (session->txn->isolation != WT_ISO_SNAPSHOT)
+        WT_RET_MSG(
+          session, ENOTSUP, "not supported in read-committed or read-uncommitted transactions.");
+
+    /* Return error if the session has performed any write operations. */
+    if (F_ISSET(session->txn, WT_TXN_HAS_ID))
+        WT_RET_MSG(session, ENOTSUP,
+          "sessions with read-committed or read-uncommitted isolation performed write operation.");
```

Comment:
The error message is wrong. We already verified the isolation level at the above check. It can be changed to `not supported in write transactions`? There is no need to mention any isolation level.

Reply:
Updated the error message.